bokehjs/src/lib/models/tools/edit/poly_edit_tool.ts

```diff
@@ -56,30 +57,48 @@ export class PolyEditToolView extends PolyToolView {
     if (!this.model.active)
       return
 
+    const vsync_renderer = this.model.renderers[0]
+    const vsync_updater = () => this._update_vertices(vsync_renderer)
+    let vsync_ds = null
+    if (vsync_renderer !== undefined) vsync_ds = vsync_renderer.data_source
```

Comment:
@mattpap Might correct me here but this might be simpler:

```suggestion
    const vsync_ds = vsync_renderer?.data_source
```

but also necessitates the change below.

Reply:
Definitively.